src/youtube.js

```diff
@@ -0,0 +1,1104 @@
+    return 'none';
+  }
+
+  /**
+   * Set if the video will automatically start playing as soon as it can do so without stopping.
+   * @param {boolean} autoplay - The autoplay value.
+   * @public
+   * @returns {void}
+   */
+  set autoplay(autoplay: boolean): void {}
+
+  /**
+   * Gets the autoplay value of the video element.
+   * @returns {boolean} - The autoplay value.
+   * @public
+   */
+  get autoplay(): boolean {
+    return false;
+  }
+
+  /**
+   * Set to specifies that the video will start over again, every time it is finished.
+   * @param {boolean} loop - the loop value.
+   * @public
+   * @returns {void}
+   */
+  set loop(loop: boolean) {}
+
+  /**
+   * Gets the loop value of the video element.
+   * @returns {boolean} - The loop value.
+   * @public
+   */
+  get loop(): boolean {
+    return false;
+  }
+
+  /**
+   * Set to specifies that video controls should be displayed.
+   * @param {boolean} controls - the controls value.
+   * @public
+   * @returns {void}
+   */
+  set controls(controls: boolean): void {}
+
+  /**
+   * Gets the controls value of the video element.
+   * @returns {boolean} - The controls value.
+   * @public
+   */
+  get controls(): boolean {
+    return false;
+  }
+
+  /**
+   * Sets the current playback speed of the audio/video.
+   * @param {Number} playbackRate - The playback speed value.
+   * @public
+   * @returns {void}
+   */
+  set playbackRate(playbackRate: number): void {
+    if (this._api) {
+      this._api.setPlaybackRate(playbackRate);
+    }
+  }
+
+  /**
+   * Gets the current playback speed of the audio/video.
+   * @returns {Number} - The current playback speed value.
+   * @public
+   */
+  get playbackRate(): number {
+    return this._api ? this._api.getPlaybackRate() : 1;
+  }
+
+  /**
+   * Sets the default playback speed of the audio/video.
+   * @param {Number} defaultPlaybackRate - The default playback speed value.
+   * @public
+   * @returns {void}
+   */
+  set defaultPlaybackRate(defaultPlaybackRate: number) {}
+
+  /**
+   * Gets the default playback speed of the audio/video.
+   * @returns {Number} - The default playback speed value.
+   * @public
+   */
+  get defaultPlaybackRate(): number {
+    return 1;
+  }
+
+  /**
+   * The ended property returns whether the playback of the audio/video has ended.
+   * @returns {boolean} - The ended value.
+   * @public
+   */
+  get ended(): boolean {
+    return this._api ? this._api.getPlayerState() === window.YT.PlayerState.ENDED : false;
+  }
+
+  /**
+   * The error property returns a MediaError object.
+   * @returns {MediaError} - The MediaError object has a code property containing the error state of the audio/video.
+   * @public
+   */
+  get error(): ?MediaError {
+    return null;
+  }
+
+  /**
+   * @returns {Number} - The current network state (activity) of the audio/video.
+   * @public
+   */
+  get networkState(): number {
+    if (!(this._api && this._api.getPlayerState)) {
+      return 0;
```

Comment:
do we have enum for the numbers?

Reply:
no, and this is currently not exposed so we can add it as a separate task